$OpenBSD: patch-stepcore_tests_worldcopy_cc,v 1.1.1.1 2013/04/24 19:06:28 zhuk Exp $
[ 98%] Building CXX object stepcore_tests/CMakeFiles/tests.dir/worldcopy.o
cd /usr/ports/pobj/step-4.7.1/build-i386/stepcore_tests && /usr/bin/c++   -DSTEPCORE_WITH_QT -DSTEPCORE_WITH_GSL -DQT_TEST_LIB -DQT_CORE_LIB -DQT_NO_DEBUG -O2 -pipe   -Woverloaded-v
irtual -fvisibility=hidden -Werror=return-type -fvisibility-inlines-hidden -O2 -DNDEBUG -DQT_NO_DEBUG -I/usr/ports/pobj/step-4.7.1/build-i386/stepcore_tests -I/usr/ports/pobj/step-4
.7.1/step-4.7.1/stepcore_tests -I/usr/ports/pobj/step-4.7.1/step-4.7.1 -I/usr/local/include/eigen2 -I/usr/local/include/X11/qt4 -I/usr/local/include/X11/qt4/QtTest -I/usr/local/incl
ude/X11/qt4/QtCore    -Wall -o CMakeFiles/tests.dir/worldcopy.o -c /usr/ports/pobj/step-4.7.1/step-4.7.1/stepcore_tests/worldcopy.cc
In file included from /usr/ports/pobj/step-4.7.1/step-4.7.1/stepcore/types.h:28,
                 from /usr/ports/pobj/step-4.7.1/step-4.7.1/stepcore/world.h:26,
                 from /usr/ports/pobj/step-4.7.1/step-4.7.1/stepcore_tests/worldcopy.cc:20:
/usr/local/include/eigen2/Eigen/StdVector:9:2: error: #error you must include <Eigen/StdVector> before <vector>. Also note that <Eigen/Sparse> includes <vector>, so it must be inclu
ded after <Eigen/StdVector> too.

--- stepcore_tests/worldcopy.cc.orig	Tue Oct 11 05:33:33 2011
+++ stepcore_tests/worldcopy.cc	Tue Oct 11 05:48:02 2011
@@ -15,11 +15,11 @@
    along with StepCore; if not, write to the Free Software
    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
+#include <stepcore/types.h>
 #include "maintest.h"
 
 #include <stepcore/world.h>
 #include <stepcore/solver.h>
-#include <stepcore/types.h>
 
 class WorldCopyTestItem: public StepCore::Item,
                          public StepCore::Body,
